### ISSUE TITLE
Create a search option in both connection page and source page 

### ISSUE THREAD

**Issue by siddhant3030** [DalgoT4D/DDP_backend#429]
LAHI currently has more than 30 + connections and a lot of sources. It would be great to have a search functionality so that the user can easily search for the connections and sources that he is looking for. 

So I looked at what Fivetran is doing and they have something like this 

<img width="1170" alt="Screenshot 2024-01-10 at 7 47 53 PM" src="https://github.com/DalgoT4D/DDP_backend/assets/9019958/dfe5f60e-be9d-4435-b486-60e229d96291">

we can even select which source we want to look at it is something like this

<img width="1172" alt="Screenshot 2024-01-10 at 7 48 57 PM" src="https://github.com/DalgoT4D/DDP_backend/assets/9019958/ad808e49-6520-404a-8db5-21e7f80f0240">

In the case of LAHI, they have Google sheet, Kobotoolbox, and Lighthouse upon choosing one of these options look at how easy it is for the USER to look at the connections for a particular source 

<img width="1173" alt="Screenshot 2024-01-10 at 7 49 04 PM" src="https://github.com/DalgoT4D/DDP_backend/assets/9019958/70acab96-1587-4ee4-ae94-9ae07c9c684a">

I believe the search option can help users find the connection for what they want and the other one is to filter based on just source.

**Comment by nairabhishek73** [comment]
Agreed! This is a great idea. @fatchat when do you think we can work on this?
cc: @ashwinit4d

**Comment by fatchat** [comment]
right now only one client has enough connections to warrant this, and they're not even a client yet

so 
1. low priority
2. after our frontend guy joins

same reasoning for 
https://github.com/DalgoT4D/DDP_backend/issues/312

and

https://github.com/DalgoT4D/DDP_backend/issues/301
